bin/regexp-tree

```diff
@@ -7,45 +7,32 @@ const fs = require('fs');
 const os = require('os');
 const regexpTree = require('..');
 
-const options = require('nomnom')
-  .script('regexp-tree')
-  .options({
-    expression: {
-      abbr: 'e',
-      help: 'A regular expression to be parsed',
-      type: 'string',
-      required: true,
-    },
-    loc: {
-      abbr: 'l',
-      help: 'Whether to capture AST node locations',
-      flag: true,
-    },
-    optimize: {
-      abbr: 'o',
-      help: 'Apply optimizer on the passed expression',
-      flag: true,
-    },
-    compat: {
-      abbr: 'c',
-      help: 'Apply compat-transpiler on the passed expression',
-      flag: true,
-    },
-    table: {
-      abbr: 't',
-      help: 'Print NFA/DFA transition tables (nfa/dfa/all)',
-      callback(table) {
-        const ALLOWED = ['nfa', 'dfa', 'all'];
-        if (!ALLOWED.includes(table)) {
-          return colors.red(
-            `\nUnknown table option: "${table}". ` +
-            `Available options are: ${ALLOWED.join('/')}.\n`
-          );
-        }
-      }
-    },
-  })
-  .parse();
+const VERSION = require('../package.json').version;
+
+const options = require('commander')
+  .version(VERSION)
+  .option(
+    '-e, --expression <expression>',
+    'A regular expression to be parsed'
+  )
+  .option(
+    '-l, --loc',
+    'Whether to capture AST node locations'
+  )
+  .option(
+    '-o, --optimize',
+    'Apply optimizer on the passed expression'
+  )
+  .option(
+    '-c, --compat',
+    'Apply compat-transpiler on the passed expression'
+  )
+  .option(
+    '-t, --table <table>',
+    'Print NFA/DFA transition tables (nfa/dfa/all)',
+    /^[nd]fa|all$/
+  )
+  .parse(process.argv);
```

Comment:
Yeah, I noticed that too, and this is unfortunate. The same goes to the `table` option which _should_ be one of the "nfa", "dfa", or "all". Even though it accepts the regexp `/^[nd]fa|all$/`, it [doesn't enforce it](https://github.com/tj/commander.js/blob/d359b75911aeaff2ed767d65b92944773691656d/index.js#L369-L374), but takes _any_ value, like "foo".

I'll add some explicit check with `console.error` in case. As to required `expression`, weird, from the docs `<expression>` is a required, and `[expression]` is optional param.

Reply:
Hm, indeed, if the value of `table` does not match the regexp it defaults to `true`.

Regarding the `<>` syntax, using it only ensures that an error is shown when using `regexp-tree -e` without actual value for the expression arg.

Maybe `commander` is not the alternative we need. I suggested it only because that's what `nomnom` suggests in its README. I'll take a quick look at other projects and will tell you if I find something that looks appropriate.

Reply:
[`yargs`](https://github.com/yargs/yargs) is quite famous and looks interesting.

The `options` method makes for a nice short syntax: https://github.com/yargs/yargs/blob/master/docs/api.md#optionskey-opt

Mandatory options can be specified with `demandOption: true`
Aliases can be specified with `alias: 'the-alias'`
I wonder if `type: 'string'` can be combined with `choices: ['nfa', 'dfa', 'all']` to limit the values of the `table` arg while allowing only one value.

Reply:
Oh yeah, we actually use `yargs` in some other internal projects at work, so I think we can switch to it; I'll update the PR later on today.

Reply:
OK, update to `yargs`.